modules/stochastic_tools/src/actions/StochasticToolsAction.C

```diff
@@ -65,24 +66,62 @@ StochasticToolsAction::act()
   }
 
   // [Problem]
-  else if (_current_task == "auto_create_problem" && getParam<bool>("auto_create_problem") &&
-           !_awh.hasActions("create_problem"))
+  else if (_current_task == "auto_create_problem" && getParam<bool>("auto_create_problem"))
   {
-    // Build the Action parameters
-    InputParameters action_params = _action_factory.getValidParams("CreateProblemAction");
+    if (_awh.hasActions("create_problem"))
+    {
+      const auto & actions = _awh.getActionListByName("create_problem");
+      for (const auto & act : actions)
+      {
+        CreateProblemAction * p = dynamic_cast<CreateProblemAction *>(act);
+        if (!p)
+          continue;
```

Comment:
Pretty sure this can be done more simply:
```suggestion
      for (auto & act : _awh.getActions<CreateProblemAction>())
```

Reply:
Yeah, the issue is that `getObjectParams()` is a non-const function and `act` here is a const pointer.